browser/permissions/chrome_permission_origin_lifetime_monitor.h

```diff
@@ -0,0 +1,43 @@
+/* Copyright (c) 2021 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_BROWSER_PERMISSIONS_CHROME_PERMISSION_ORIGIN_LIFETIME_MONITOR_H_
+#define BRAVE_BROWSER_PERMISSIONS_CHROME_PERMISSION_ORIGIN_LIFETIME_MONITOR_H_
+
+#include "base/containers/flat_set.h"
+#include "base/memory/weak_ptr.h"
+#include "brave/components/permissions/permission_origin_lifetime_monitor.h"
+
+class Profile;
+
+namespace permissions {
+
+class ChromePermissionOriginLifetimeMonitor
```

Comment:
Brave is also an embedder, so I' suggest to change Chrome to Brave :)
Also it seems there is nothing specific to permissions, so this probably can be a general tool for other "ephemeral" stuff. Maybe worth renaming